### DISCUSSION TITLE
Cannot access Flutter from Android Studio's Terminal

### DISCUSSION THREAD

**Discussion by ste8** [pingbird/puro#41] [Q&A] [answered]
This is actually more a question about Android Studio rather than Puro, but I have encountered this problem as a result of using Puro, so I'll try to ask here.
I have a new MacBook and instead of installing Flutter in the usual way, I wanted to try Puro (so I have now installed Flutter and Dart only through Puro).
In the preferences of Android Studio I can see Dart SDK and Flutter SDK path set correctly, respectively as 
/Users/{username}/.puro/envs/{environment}/flutter/bin/cache/dart-sdk
/Users/{username}/.puro/envs/{environment}/flutter
and I can run commands such as pub get using Android Studio's interface.

But if I try to execute any flutter command (e.g. 'flutter doctor') in Android Studio's Terminal, I receive the error "zsh: command not found: flutter" because the Terminal cannot see Flutter's SDK path.

The interesting thing is that Visual Studio Code, instead, has an option (enabled by default) called "Dart: Add Sdk to Terminal Path" that allow to call flutter commands from Terminal. And here they work!

I tried to search some info about Android Studio, but I haven't been able to make it work.
In a project I am working on, a makefile was defined with several references to flutter's commands, and I cannot execute it from Android Studio's terminal due to this issue.
A workaround is to execute it from VS Code, but I'd prefer to have everything inside Android Studio.

Any suggestion about this issue?

**Comment by pingbird** [comment]
It should be automatically added to your PATH, ensure:

1. The environment you want to use is set as the global default e.g. `puro use -g stable`
2. You are using zsh as your shell for Android Studio, `echo $SHELL` should print `zsh`
3. `~/.zshrc` contains `export PATH="$PATH:` lines that the puro installer should've added automatically
4. Nothing else is overriding the PATH variable, such as an `export` command that doesn't prepend w/ `$PATH:` or Android Studio settings overriding it in `Tools > Terminal > Environment Variables`

**Comment by ste8** [comment]
Thanks for the answer!
After setting the environment as the global default (puro use -g) it works, but in this way I have to remember to switch the global default everytime I change project (with a different environment), right?
With Visual Studio Code this step isn't needed.

**Reply by pingbird** [reply, marked as answer]
To use the project-specific flutter version you need to use the subcommand `puro flutter` or `puro dart`, create an issue on the flutter-intellij repo if you want a similar option as VSCode's "Add Sdk to Terminal Path".

**Reply by tksoh** [reply]
@ste8 I use VS Code. With puro, I find myself better off with the "Add Sdk to Terminal Path" option disabled. Relying on `puro use -g`  would probably turn into a nightmare quickly if you have multiple projects using different puro env in each one.

As a quick workaround, perhaps you can consider creating a shell script named `flutter` to call `puro flutter`, and put it in puro's bin directory (since it's already in the shell PATH). This is what I did, although my script is slightly more sophisticated, in order to get around `flutter doctor` warning messages on different SDK on the path.

Just some thought.

**Reply by pingbird** [reply]
That's how things used to work, but it broke InteliJ's SDK location detection so we switched to having the default environment part of the PATH
